source/common/websocket/codec.cc

```diff
@@ -0,0 +1,220 @@
+    return false;
+  }
+  frame_.opcode_ = opcode;
+  frame_.final_fragment_ = data.front() & 0x80;
+  state_ = State::FrameHeaderMaskFlagAndLength;
+  return 1;
+}
+
+uint8_t Decoder::doDecodeMaskFlagAndLength(absl::Span<const uint8_t>& data) {
+  frameMaskFlag(data.front());
+  if (length_ == 0x7e) {
+    num_remaining_extended_length_bytes_ = kPayloadLength16Bit;
+    length_ = 0;
+    state_ = State::FrameHeaderExtendedLength;
+  } else if (length_ == 0x7f) {
+    num_remaining_extended_length_bytes_ = kPayloadLength64Bit;
+    length_ = 0;
+    state_ = State::FrameHeaderExtendedLength;
+  } else if (num_remaining_masking_key_bytes_ > 0) {
+    state_ = State::FrameHeaderMaskingKey;
+  } else {
+    frameDataStart();
+  }
+  return 1;
+}
+
+uint8_t Decoder::doDecodeExtendedLength(absl::Span<const uint8_t>& data) {
+  uint8_t bytes_decoded = 0;
+  uint64_t bytes_to_decode = data.length() <= num_remaining_extended_length_bytes_
+                                 ? data.length()
+                                 : num_remaining_extended_length_bytes_;
+  while (bytes_to_decode > 0) {
+    length_ |= static_cast<uint64_t>(data.at(bytes_decoded))
+               << 8 * (num_remaining_extended_length_bytes_ - 1);
+    num_remaining_extended_length_bytes_--;
+    bytes_to_decode--;
+    bytes_decoded++;
+  }
+
+  if (num_remaining_extended_length_bytes_ == 0) {
+    if (num_remaining_masking_key_bytes_ > 0) {
+      state_ = State::FrameHeaderMaskingKey;
+    } else {
+      frameDataStart();
+    }
+  }
+  return bytes_decoded;
+}
+
+uint8_t Decoder::doDecodeMaskingKey(absl::Span<const uint8_t>& data) {
+  if (!frame_.masking_key_.has_value()) {
+    frame_.masking_key_ = 0;
+  }
+  uint8_t bytes_decoded = 0;
+  uint64_t bytes_to_decode = data.length() <= num_remaining_masking_key_bytes_
+                                 ? data.length()
+                                 : num_remaining_masking_key_bytes_;
+  while (bytes_to_decode > 0) {
+    frame_.masking_key_.value() |= static_cast<uint32_t>(data.at(bytes_decoded))
+                                   << 8 * (num_remaining_masking_key_bytes_ - 1);
+    num_remaining_masking_key_bytes_--;
+    bytes_to_decode--;
+    bytes_decoded++;
+  }
+  if (num_remaining_masking_key_bytes_ == 0) {
+    frameDataStart();
+  }
+  return bytes_decoded;
+}
+
+uint64_t Decoder::doDecodePayload(absl::Span<const uint8_t>& data) {
+  uint64_t remain_in_buffer = data.length();
+  uint64_t bytes_decoded = 0;
+  if (remain_in_buffer <= length_) {
+    frameData(data.data(), remain_in_buffer);
+    bytes_decoded += remain_in_buffer;
+    length_ -= remain_in_buffer;
+  } else {
+    frameData(data.data(), length_);
+    bytes_decoded += length_;
+    length_ = 0;
+  }
+  if (length_ == 0) {
+    state_ = State::FrameFinished;
+  }
+  return bytes_decoded;
+}
+
+absl::optional<std::vector<Frame>> Decoder::decode(Buffer::Instance& input) {
+  absl::optional<std::vector<Frame>> output = absl::nullopt;
```

Comment:
Let's initialize this to a valid vector here 
```
absl::optional<std::vector<Frame>> output(std::vector<Frame>());
```
Then we can remove the corresponding code from `frameDataEnd()`;